src/Orleans/Streams/PersistentStreams/PersistentStreamProvider.cs

```diff
@@ -99,19 +110,48 @@ public class PersistentStreamProvider<TAdapterFactory> : IInternalStreamProvider
                 maxEventDeliveryTime = ConfigUtilities.ParseTimeSpan(timeout,
                     "Invalid time value for the " + MAX_EVENT_DELIVERY_TIME + " property in the provider config values.");
 
-            logger.Info("Initialized PersistentStreamProvider<{0}> with name {1}, {2} = {3}, {4} = {5} and with Adapter {6}.",
+            string startup;
+            if (config.Properties.TryGetValue(STARTUP_STATE, out startup))
+            {
+                if(!Enum.TryParse(startup, true, out startupState))
+                    throw new ArgumentException(
+                        String.Format("Unsupported value '{0}' for configuration parameter {1} of stream provider {2}.", startup, STARTUP_STATE, config.Name));
+            }
+            else 
+                startupState = StreamProviderStartupState.Started;
+
+            logger.Info("Initialized PersistentStreamProvider<{0}> with name {1}, {2} = {3}, {4} = {5}, {6} = {7} and with Adapter {8}.",
                 typeof(TAdapterFactory).Name, Name, 
                 GET_QUEUE_MESSAGES_TIMER_PERIOD, getQueueMsgsTimerPeriod,
-                INIT_QUEUE_TIMEOUT, this.initQueueTimeout, 
+                INIT_QUEUE_TIMEOUT, initQueueTimeout,
+                STARTUP_STATE, startupState,
                 queueAdapter.Name);
         }
 
         public async Task Start()
         {
-            if (queueAdapter.Direction.Equals(StreamProviderDirection.ReadOnly) ||
-                queueAdapter.Direction.Equals(StreamProviderDirection.ReadWrite))
+            if (providerRuntime.InSilo) 
```

Comment:
Why do we care if this is in a silo or not?  It really depends on the runtime right?  If the runtime is a ISiloSideStreamProviderRuntime the we can continue.  With this in mind, why not just get the siloRuntime as we do below then null check it.  This is less code and ensures that the siloRuntime is not null, which the current code does not.